### ISSUE TITLE
M2.1.1 : Re-saving a product attribute with a different name than it's code results in an error

### ISSUE THREAD

**Issue by kanduvisla** [magento/magento2#6770]
### Preconditions
1. Magento 2.1.1
2. I have product attribute of the type 'dropdown', which name is 'Afmetingen uitwendig (BxDxH) in mm' and it's code is 'afmetingen_uitwendig'.
### Steps to reproduce
1. Add a couple of options
2. Save the attribute
3. Add a couple of options more
4. Re-save the attribute
### Expected result
1. Options should be added
2. No errors should be shown
### Actual result

Upon re-saving, Magento tries to re-create the attribute code:

![screen shot 2016-09-27 at 15 33 34](https://cloud.githubusercontent.com/assets/346681/18875894/ec1525ae-84c7-11e6-9dc7-37e56e3e3a08.png)

As you can see, it seems that the existing attribute code is ignored. Further inspection of the form confirms this:

![screen shot 2016-09-27 at 15 34 19](https://cloud.githubusercontent.com/assets/346681/18875900/f1eeb468-84c7-11e6-8c48-29a4b1b16780.png)

![screen shot 2016-09-27 at 15 34 27](https://cloud.githubusercontent.com/assets/346681/18875906/f59f6008-84c7-11e6-9bb6-34b39732ec37.png)

The problem here is obvious: setting the form to `disabled="disabled"` disables the form value from submitting. You can check this in the form data send in the request. The field `attribute_code` is missing here, thus Magento will generate a new attribute code.

The solution in my opinion is to change `disabled="disabled"` to `readonly="readonly"`. This makes sure the field isn't editable, but it's value is still submitted.

**Comment by sevos1984** [comment]
Problem is in attribute code length that can't be more than 30 symbols. If add code manually all works fine. I've create internal ticket for that MAGETWO-59153. Thanks for posting.

**Comment by kanduvisla** [comment]
This error still persists in M2.1.2. Is there any sight on when this is going to be fixed?

**Comment by kanduvisla** [comment]
I've created a pull request that fixes this issue. What it does:

```
$attributeId = $this->getRequest()->getParam('attribute_id');

$attributeCode = $attributeId ?
    $this->attributeRepository->get($attributeId)->getAttributeCode() :
    $this->getRequest()->getParam('attribute_code');

if (!$attributeCode) {
    $attributeCode = $this->generateCode($this->getRequest()->getParam('frontend_label')[0]);
}
```

In other words:

When there is an attribute ID (which is the case with existing attributes), load the already known attribute code. If there is no attribute ID, try the 'old behaviour'. Which is: check for a param called `attribute_code` or try to generate a code according to the `frontend_label`.

**Comment by magento-engcom-team** [comment]
@kanduvisla, thank you for your report.
We've created internal ticket(s) MAGETWO-59153 to track progress on the issue.

**Comment by raumatbel** [comment]
@magento-engcom-team can you assign me please?

**Comment by okorshenko** [comment]
@kanduvisla  The issue has been fixed and delivered in 2.2-develop branch. Will be available with upcoming patch release.
